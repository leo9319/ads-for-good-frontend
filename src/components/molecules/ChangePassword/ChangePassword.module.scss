@import '../UserProfile/UserProfile.scss';

.formLayout {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  align-self: stretch;
  gap: var(--spacing-lg, 24px);
}

.headingSection {
  font-weight: var(--typography-font-weight-semibold) !important;
  font-size: var(--typography-font-size-4) !important;
  line-height: var(--typography-line-height-4) !important;
}

.passwordButton {
  @extend .userProfileEditBtn;
}
