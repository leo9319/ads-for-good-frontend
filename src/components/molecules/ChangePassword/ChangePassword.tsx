import React, { ReactElement, useEffect, useState } from 'react';
import Box from '@radix-styles/atoms/Box';
import Button from '@radix-styles/atoms/Button';
import Heading from '@radix-styles/atoms/Heading';
import PasswordVerificationForm from '../PasswordVerificationForm';
import SmallCard from '../Cards/SmallCard';
import { PasswordVerificationFormProps } from '../PasswordVerificationForm/PasswordVerificationForm';

import styles from './ChangePassword.module.scss';
import { AnalyticsHandler } from '@utils/Analytics';

/**
 * @interface
 * The Form props type of {@link ChangePassword}.
 */
export interface ChangePasswordProps extends PasswordVerificationFormProps {
  /**
   * Form status prop to check the status of form after submitted
   *  @property
   * */
  formSubmitStatus?: boolean;
  /**
   * Lable for form title
   *  @property
   * */
  title: string;
}

/**
 * ChangePassword component
 *
 * @group Molecules
 * @category Component
 *
 * @param {ChangePasswordProps} props - The ChangePassword props
 * @returns {ReactElement} - ChangePassword Component
 */
export const ChangePassword = ({
  title,
  currentPasswordLabel = '',
  currentPasswordPlaceholder = '',
  passwordLabel,
  confirmPasswordLabel,
  passwordPlaceholder,
  reEnterpasswordPlaceholder,
  buttonLabel = '',
  formError,
  isSignup = false,
  onSubmit,
  onCancel,
  analyticsProps,
  isUserprofile = false,
  formSubmitStatus = false,
  loading = false,
}: ChangePasswordProps): ReactElement => {
  const [isEdit, setIsEdit] = useState(false);
  const { trackClick } = AnalyticsHandler();

  useEffect(() => {
    setIsEdit(formSubmitStatus);
  }, [formSubmitStatus]);

  const handleCancel = () => {
    setIsEdit(false);
    if (onCancel) onCancel();
  };
  const handleEdit = () => {
    setIsEdit(true);
    trackClick({
      target: '',
      location: 'Password',
      name: 'Password-Edit',
      type: 'button',
      contentTypeOrPosition: '',
    });
  };
  return (
    <SmallCard size="3" mode="inverse">
      <Box className={styles.formLayout}>
        <Heading className={styles.headingSection}>{title}</Heading>
        {isEdit ? (
          <PasswordVerificationForm
            currentPasswordLabel={currentPasswordLabel}
            currentPasswordPlaceholder={currentPasswordPlaceholder}
            isUserprofile={isUserprofile}
            passwordLabel={passwordLabel}
            passwordPlaceholder={passwordPlaceholder}
            confirmPasswordLabel={confirmPasswordLabel}
            reEnterpasswordPlaceholder={reEnterpasswordPlaceholder}
            buttonLabel={buttonLabel}
            onSubmit={onSubmit}
            onCancel={handleCancel}
            analyticsProps={analyticsProps}
            formError={formError}
            isSignup={isSignup}
            loading={loading}
          />
        ) : (
          <Button
            className={styles.passwordButton}
            text="Edit"
            mode="secondary"
            type="button"
            disabled={false}
            onClick={handleEdit}
            iconLeft={true}
            icon="icon-edit"
            size="lg"
          />
        )}
      </Box>
    </SmallCard>
  );
};

export default ChangePassword;
