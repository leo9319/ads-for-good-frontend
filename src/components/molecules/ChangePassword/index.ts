export type { ChangePasswordProps } from './ChangePassword';
export { default } from './ChangePassword';
export { ChangePassword } from './ChangePassword';
