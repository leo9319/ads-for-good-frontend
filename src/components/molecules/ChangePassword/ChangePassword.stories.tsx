import React from 'react';
import ChangePassword from './ChangePassword';
import { Meta, StoryObj } from '@storybook/react';
import Theme from '@radix-styles/atoms/Theme';

const meta = {
  title: 'Library/Molecules/ChangePassword',
  component: ChangePassword,
  decorators: [
    Story => (
      <Theme>
        <Story />
      </Theme>
    ),
  ],
  parameters: {
    layout: 'fullscreen',
  },
  tags: ['autodocs'],
  argTypes: {},
} satisfies Meta<typeof ChangePassword>;

export default meta;

type Story = StoryObj<typeof meta>;

export const ChangePasswordSection: Story = {
  args: {
    title: 'Password',
    currentPasswordLabel: 'Current Password',
    currentPasswordPlaceholder: 'Current Password',
    isUserprofile: true,
    passwordLabel: 'New Password',
    passwordPlaceholder: 'New Password',
    confirmPasswordLabel: 'Confirm New Password',
    reEnterpasswordPlaceholder: 'Confirm New Password',
    buttonLabel: 'Submit',
    onSubmit: () => console.log('submitted'),
    onCancel: () => console.log('cancelled'),
    analyticsProps: { form: 'changepassword', type: 'changepassword' },
    formSubmitStatus: false,
  },
};
