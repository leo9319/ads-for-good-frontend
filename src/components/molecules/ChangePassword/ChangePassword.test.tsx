import React from 'react';
import { render } from '@testing-library/react';
import { ChangePassword } from './ChangePassword';

describe('ChangePassword', () => {
  it('Should render successfully', () => {
    const { baseElement } = render(
      <ChangePassword
        title="Password"
        currentPasswordLabel="Current Password"
        currentPasswordPlaceholder="Current Password"
        isUserprofile={true}
        passwordLabel="New Password"
        passwordPlaceholder="New Password"
        confirmPasswordLabel="Confirm New Password"
        reEnterpasswordPlaceholder="Confirm New Password"
        buttonLabel="Submit"
        onSubmit={() => console.log('submitted')}
        onCancel={() => console.log('cancelled')}
        analyticsProps={{ form: 'changepassword', type: 'changepassword' }}
        formSubmitStatus={false}
      />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
