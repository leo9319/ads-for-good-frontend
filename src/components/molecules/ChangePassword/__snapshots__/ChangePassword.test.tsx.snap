// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassword Should render successfully 1`] = `
<body>
  <div>
    <div
      class="rt-reset rt-BaseCard rt-Card rt-r-size-1 rt-variant-surface container size-3 inverse"
    >
      <div
        class="rt-Box formLayout"
      >
        <h1
          class="rt-Heading rt-r-size-6 headingContainer mocked-class-name typography-font-family-default headingSection"
          id=""
        >
          Password
        </h1>
        <div
          class="rt-Flex rt-r-display-flex rt-r-fd-row rt-r-gap-3 buttonGroupContainer passwordButtonGroup"
        >
          <button
            aria-label="Edit"
            class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button wvButton xl secondary none rt-Box buttonWrapper"
            data-accent-color=""
            type="button"
          >
            <div
              class="buttonContent iconLeft"
            >
              <span>
                Edit
              </span>
              <span
                class="icon"
                style="border-radius: 0; display: grid; align-items: center; justify-content: center;"
              >
                <span
                  class="icon icon-edit"
                />
              </span>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
